src/saving_files.jl

```diff
@@ -24,6 +24,8 @@ end
 ```
 
 ## Keywords
+* `filename = savename(prefix, config, suffix; kwargs...)` : Name of the file
```

Comment:
It is problematic that this keyword is `filename` but also `filename` is the one with `joinpath(path, filename)`. We need to either state `filename` with a different name in the beginning of the documentation string, or use a different name for the keyword. 

Kind of off-topic, but I'd strongly recommend you to not include paths in `filename` and use the `path` argument explicitly. I used to do the same thing as you when we started developing DrWatson, but the other lead devs presented very straightforward arguments why one shouldn't do this. I don't remember these arguments anymore, they were something about "it doesn't work for all OS" or something. In any case, it isn't good practice.

Reply:
Yeah, I actually had the same concern about `filename`... It's confusing to have it in two places with a slightly different meaning.

In https://github.com/JuliaDynamics/DrWatson.jl/pull/318#issuecomment-1013953382, we discussed whether the return value should be called `filename`, or just `file`. Maybe we should revisit that, and change it to `file`? That would mean we can use

~~~
file = joinpath(path, filename)
~~~

And, in the documentation, we would write

~~~
produce_or_load([path="",] config, f; kwargs...) -> data, file
~~~

(where `kwargs` includes `filename`), which would be much cleaner.

I also agree with you that keeping `filename` and `path` separate is generally a good idea.

Reply:
okay, let's go with `filename` for the name without the path and `file` for the joinpath-ed version.